File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        if n == 1:return 1
        for i in range(n):
            if sum(range(1,i+1)) == sum(range(i,n+1)): 
                # +1 is added bcozz.. it ranges from 1,i+1 it doesnt include i+1 
                return i
        return -1 

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        if(n== 1){
            return 1;
        }
        int left_sum = 0;
        int right_sum = (n*(n+1)/2);
        for(int i=0;i<=n;i++){
            left_sum+= i;
            right_sum-=i;
            if(left_sum == right_sum+i){
                return i;
            }
        }
        return -1;
    }
}